packages/app/ui/components/Wayfinding/SplashSequence.tsx

```diff
@@ -0,0 +1,492 @@
+import { Button, Icon, Text, triggerHaptic } from '@tloncorp/ui';
+import React, {
+  ComponentProps,
+  PropsWithChildren,
+  useCallback,
+  useMemo,
+} from 'react';
+import { Dimensions, Image } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import {
+  ColorTokens,
+  View,
+  XStack,
+  YStack,
+  ZStack,
+  isWeb,
+  styled,
+} from 'tamagui';
+
+import { useActiveTheme } from '../../../provider';
+import { useStore } from '../../contexts';
+import { ListItem } from '../ListItem';
+import { Squiggle } from '../Squiggle';
+import { PrivacyThumbprint } from './visuals/PrivacyThumbprint';
+
+enum SplashPane {
+  Welcome = 'welcome',
+  Group = 'Group',
+  Channels = 'Channels',
+  Privacy = 'Privacy',
+  Invite = 'Invite',
+}
+
+function SplashSequenceComponent(props: { onCompleted: () => void }) {
+  const insets = useSafeAreaInsets();
+  const store = useStore();
+  const [currentPane, setCurrentPane] = React.useState<SplashPane>(
+    SplashPane.Welcome
+  );
+
+  const handleSplashCompleted = useCallback(() => {
+    store.completeWayfindingSplash();
+    props.onCompleted();
+  }, [props, store]);
+
+  return (
+    <View
+      flex={1}
+      // marginTop={insets.top}
+      // marginBottom={insets.bottom}
+      // paddingTop="$2xl"
+      // paddingBottom="$xl"
+    >
+      {currentPane === 'welcome' && (
+        <WelcomePane onActionPress={() => setCurrentPane(SplashPane.Group)} />
+      )}
+      {currentPane === 'Group' && (
+        <GroupsPane onActionPress={() => setCurrentPane(SplashPane.Channels)} />
+      )}
+      {currentPane === 'Channels' && (
+        <ChannelsPane
+          onActionPress={() => setCurrentPane(SplashPane.Privacy)}
+        />
+      )}
+      {currentPane === 'Privacy' && (
+        <PrivacyPane onActionPress={() => setCurrentPane(SplashPane.Invite)} />
+      )}
+      {currentPane === 'Invite' && (
+        <InvitePane onActionPress={handleSplashCompleted} />
+      )}
```

Comment:
is it difficult to use react navigation here? just curious because I know we use this pattern elsewhere too

Reply:
yeah, good question. I generally do it if a feature/story feels "self contained" and splitting into separate screens seems "too heavy". You do however lose the transition animations. Using react navigation would complicate the desktop modal, but I don't see other significant drawbacks.

Open to refactoring if you'd prefer?